# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/tests/browser/events/browser_test_scrolling.js
# Commit: d7616232740d
# Full Hash: d7616232740d98904aa4ef6fc6940308faec0846
# Author: Nathan LaPre <nlapre@mozilla.com>
# Date: 2023-06-08 09:27:30
# Regressor Bug: 1789235
# File Overlap Count: 1
# Description:
#   Bug 1789235: Fire a scrolling start event for anchor jump if focus in doc, r=Jamie
#   
#   This revision modifies NotifyOfAnchorJump in order to ensure that we fire a
#   scrolling start event for AT clients. Without this, clients might miss anchor
#   jump updates, since anchor jumps can arrive after getting focus. This revision
# ==============================================================================

diff -r 8d2b1e733b50 -r d7616232740d accessible/tests/browser/events/browser_test_scrolling.js
--- a/accessible/tests/browser/events/browser_test_scrolling.js	Wed Jun 07 23:24:22 2023 +0000
+++ b/accessible/tests/browser/events/browser_test_scrolling.js	Thu Jun 08 00:06:39 2023 +0000
@@ -111,3 +111,20 @@
     await onScrolling;
   }
 );
+
+// Verify that the scrolling start event is fired for an anchor change.
+addAccessibleTask(
+  `
+    <p>a</p>
+    <p>b</p>
+    <p id="c">c</p>
+  `,
+  async function (browser, accDoc) {
+    let onScrollingStart = waitForEvent(EVENT_SCROLLING_START, "c");
+    await SpecialPowers.spawn(browser, [], () => {
+      content.location.hash = "#c";
+    });
+    await onScrollingStart;
+  },
+  { chrome: true, topLevel: true }
+);
